website/static/js/nodesPrivacy.js

```diff
@@ -0,0 +1,334 @@
+//    /**
+//     * patche all the nodes in a changed state.
+//     * we will use API V2 bulk requests when it becomes available
+//     */
+//    var promise;
+//    for (var key in nodes) {
+//        var node = nodes[key];
+//        var url = API_BASE + node.id + '/';
+//        nodesPatch = []
+//        nodesPatch.push({
+//        JSON.stringify(
+//            {
+//                'data': {
+//                    'type': 'nodes',
+//                    'id': node.id,
+//                    'attributes': {
+//                        'public': node.public
+//                    }
+//                }
+//            })}
+//        });
+//    }
+//
+//        promise = $.when(promise,
+//            $.ajax({
+//                url: url,
+//                type: 'PATCH',
+//                dataType: 'json',
+//                contentType: 'application/json',
+//                crossOrigin: true,
+//                xhrFields: {withCredentials: true},
+//                processData: false,
+//                data: JSON.stringify(
+//                    {
+//                        'data': {
+//                            'type': 'nodes',
+//                            'id': node.id,
+//                            'attributes': {
+//                                'public': node.public
+//                            }
+//                        }
+//                    })
+//            }).done(function (response) {
+//            }).fail(function (xhr, status, error) {
+//                $osf.growl('Error', 'Unable to update Projects and/or Components');
+//                Raven.captureMessage('Could not PATCH project settings.', {
+//                    url: url, status: status, error: error
+//                });
+//            }));
+//
+//
+//    return promise;
+//
+//}
+
+function patchNodesPrivacy(nodes) {
+    /**
+     * patche all the nodes in a changed state.
+     * we will use API V2 bulk requests when it becomes available
+     */
+    var promise;
+    for (var key in nodes) {
+        var node = nodes[key];
+        var url = API_BASE + node.id + '/';
+
+        promise = $.when(promise,
+            $.ajax({
+                url: url,
+                type: 'PATCH',
+                dataType: 'json',
+                contentType: 'application/json',
+                crossOrigin: true,
+                xhrFields: {withCredentials: true},
+                processData: false,
+                data: JSON.stringify(
+                    {
+                        'data': {
+                            'type': 'nodes',
+                            'id': node.id,
+                            'attributes': {
+                                'public': node.public
+                            }
+                        }
+                    })
+            }).done(function (response) {
+            }).fail(function (xhr, status, error) {
+                $osf.growl('Error', 'Unable to update Projects and/or Components');
```

Comment:
Since this is in a loop, we run the risk of displaying too many growl errors. I think only one error message should be displayed if there is a failure in any of the requests.